cborg/src/Codec/CBOR/Decoding.hs

```diff
@@ -335,20 +337,19 @@ toWord64 n = W64# (wordToWord64# n)
 toInt64  n = I64# n
 toWord64 n = W64# n
 #endif
-#else
-toInt64  n = I64# (intToInt64# n)
-toWord64 n = W64# (wordToWord64# n)
```

Comment:
Why are we deleting these rather than fixing them? It's breaking the pattern used below. And it also means we cannot support multiple compiler versions that need differences in these functions.

Surely we should just use CPP here to change the impl of `toWord64` and `toInt64` so that with older GHC versions they're implemented as now, and with newer versions (where the unboxed integer types changed) to use the new implementation (that's inlined at the use site below) of
```
toInt64  n = I64# n
toWord64 n = W64# n
```

Reply:
I think I just removed these compat shims as they were not strictly necessary, but I will add them back for consistency and future-proofing.